dev/Repeater/ViewportManagerWithPlatformFeatures.cpp

```diff
@@ -381,7 +381,28 @@ void ViewportManagerWithPlatformFeatures::OnEffectiveViewportChanged(winrt::Fram
 {
     assert(!m_managingViewportDisabled);
     REPEATER_TRACE_INFO(L"%ls: \tEffectiveViewportChanged event callback \n", GetLayoutId().data());
-    UpdateViewport(args.EffectiveViewport());
+    const auto effective = args.EffectiveViewport();
```

Comment:
>const auto effective = args.EffectiveViewport(); [](start = 4, length = 48)

are you intending to check this debug code in?

Reply:
I will remove it once everything is settled, I wanted to keep it for now since otherwise it would be a bit difficult to check whether the values are correct or if I am doing something wrong.

Reply:
> @ranjeshj I'm not sure if we should take this change as an improvement or rejecting until we plan out how we want this to look in the end?

I think this needs a bit more tuning/testing.

Reply:
@ranjeshj should we close this PR then?

Reply:
@chingucoding Lets close this PR for now, but please save this work in a branch. I think this needs more discussions on the heuristic and lots of testing to ensure we don't regress in the common cases. It would be better to come up with something that works well for all the scenarios instead of changing this multiple times.